gitignore_tidy/core.py

```diff
@@ -1,123 +1,184 @@
+import dataclasses
+import itertools
 import pathlib
 import re
 import typing
+from functools import cached_property
 
-import typer
+from gitignore_tidy.logging import logger
 
 
 def tidy_file(path: pathlib.Path, *, allow_leading_whitespace: bool = False):
-
-    if not path.exists():
-        raise FileNotFoundError(f"{path} not found.")
-
-    with path.open("r") as f:
-        lines = f.read().splitlines()
-
+    lines = PlainLines.from_file(path)
     if len(lines) < 1:
-        typer.echo(f"{path} is empty.")
+        logger.info(f"File {path} is empty, not writing.")
```

Comment:
Using f-strings in logging statements is discouraged. See https://docs.astral.sh/ruff/rules/logging-f-string/

Reply:
I see. Here, it's not performance critical. I'd like to use a key-value matching, but seems not so easy with `extra`? 
I.e. without a `basicConfig`, I can't `logger.info('stuff %(key)s', extra=dict(key='my-key'))`, so I'd have to fall back on less legible, at least with many arguments, IMO `logger.info("File %s is empty, not writing.", path)`. Is that what you suggest?

Reply:
Exactly, it's definitely less readable, but still the recommended way to go AFAIK.